lib/apartment/adapters/mysql2_adapter.rb

```diff
@@ -39,6 +39,8 @@ def initialize(config)
       #   Reset current tenant to the default_tenant
       #
       def reset
+        return if @default_tenant.nil?
```

Comment:
Is this the only reason for `@default_tenant` to be nil? Wouldn't it make more sense to not call reset on the appropriate places?

Reply:
Independently of that, if default_tentant is nil, the mysql statement generated would be invalid

```
mysql> use ``
ERROR 1046 (3D000): No database selected
```

Reply:
Right, but `default_tenant` (the method) falls back on the configured default tenant. Which is why I think it was supposed to be a valid case to have `@default_tenant` to be nil.

Reply:
rightly spotted. I meant `default_tenant` not `@default_tenant`. let me double check it still works on my test app

Reply:
you are totally right

Reply:
So, I've dug further into active record and for mysql it explicitly calls establish connection without database.
I can't see any other way around it.
Also we have a todo in the code base mentioning that `public` is a postgres specific, so I'd say it's safe to assume that there should not be a default to mysql